test/integration/bucketfs_test.py

```diff
@@ -0,0 +1,170 @@
+    response = requests.put(url, data=content, auth=auth)
+    response.raise_for_status()
+    return filename, url
+
+
+def _delete_file(service, bucket, username, password, filename):
+    auth = HTTPBasicAuth(username, password)
+    url = f"{service.rstrip('/')}/{bucket}/{filename}"
+    response = requests.delete(url, auth=auth)
+    response.raise_for_status()
+    return filename, url
+
+
+@pytest.mark.parametrize(
+    "url,expected",
+    [
+        ("http://127.0.0.1:6666", {"default", "myudfs", "jdbc_adapter"}),
+    ],
+)
+def test_list_buckets(url, port, expected):
+    service = Service(url, port)
+    actual = {bucket for bucket in service}
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "bucket,url,username,password,expected",
+    [
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            {
+                "EXAClusterOS/ScriptLanguages-standard-EXASOL-7.1.0-slc-v4.0.0-CM4RWW6R.tar.gz",
+            },
+        ),
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "r",
+            "read",
+            {
+                "EXAClusterOS/ScriptLanguages-standard-EXASOL-7.1.0-slc-v4.0.0-CM4RWW6R.tar.gz",
+            },
+        ),
+        (
+            "myudfs",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            set(),
+        ),
+        (
+            "jdbc_adapter",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            set(),
+        ),
+    ],
+)
+def test_list_buckets(bucket, url, username, password, expected):
+    service = Bucket(bucket, url, username, password)
+    actual = {bucket for bucket in service}
+    assert expected.issubset(actual)
+
+
+@pytest.mark.parametrize(
+    "bucket,url,username,password,data",
+    [
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            bytes([65, 66, 67, 68, 69, 70]),
+        ),
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            [b"12", b"34", b"56", b"78"],
+        ),
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            (b"1" for _ in range(0, 10)),
+        ),
+    ],
+)
+def test_upload_to_bucket(bucket, url, username, password, data):
+    upload_file = "Uploaded-File-{rnd}.bin".format(
+        rnd="".join(random.choice(string.hexdigits) for _ in range(0, 10))
+    )
+    bucket = Bucket(bucket, url, username, password)
+
+    # make sure this file does not exist yet in the bucket
+    assert upload_file not in bucket.files
+
+    # run test scenario
+    bucket.upload(upload_file, data)
+
+    # assert expectations
+    assert upload_file in bucket.files
+
+    # cleanup
+    _, _ = _delete_file(url, bucket.name, username, password, upload_file)
```

Comment:
I suggest, a try finally with the delete around the upload, or a function scope fixture, which provides the name and cleans it up